lib/streamlit/caching/memo_decorator.py

```diff
@@ -333,10 +333,16 @@ def __call__(
                 f"Unsupported persist option '{persist}'. Valid values are 'disk' or None."
             )
 
-        # Support passing the params via function decorator, e.g.
-        # @st.memo(persist=True, show_spinner=False)
-        if func is None:
-            return lambda f: create_cache_wrapper(
+        def wrapper(f):
+            # We use wrapper function here instead of lambda function to be able to log
+            # warning in case both persist="disk" and ttl parameters specified
+            if persist == "disk" and ttl is not None:
+                _LOGGER.warning(
+                    'The decorator uses the persist="disk" and ttl '
+                    "parameters at the same time. Persist cache currently "
+                    "does not support the TTL cache feature."
+                )
+            return create_cache_wrapper(
```

Comment:
I think we should update the warning to:
- Contain the name of the annotated function
- Be clear that TTL will be ignored but `persist="disk"` won't be. Something like (cc @jrieke):

```
"The memoized function {func_name} has a TTL that will be ignored. Persistent memo caches currently don't support TTL.
```

Reply:
Tim's suggestion sounds good, let's use that!